src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -440,9 +440,63 @@ void reset() {
   /** Normalize the column name, given with the inserted row, to send to server side. */
   static String formatColumnName(String columnName) {
     Utils.assertStringNotNullOrEmpty("invalid column name", columnName);
-    return (columnName.charAt(0) == '"' && columnName.charAt(columnName.length() - 1) == '"')
-        ? columnName
-        : columnName.toUpperCase();
+    if (columnName.charAt(0) == '"' && columnName.charAt(columnName.length() - 1) == '"') {
```

Comment:
Similar to `unquote`, Do you think we have any existing function at server side for `formatColumnName` that could be reused here?

Reply:
TBH I think the problem is that server side does not return 'unquoted' literal that is stored internally.
Otherwise, the bellow `unquoteColumnName` already reflects the UI column identifier [doc](https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html).
Except for the `QUOTED_IDENTIFIERS_IGNORE_CASE ` parameter that we can also add to channel builder if needed or send from server if it is not per session.

I think we should send the 'unquoted' literal that is stored internally long term in the open channel response
then we do not need this `formatColumnName` at all, just always apply `unquoteColumnName` to user input.

Now we can solve this by applying `unquoteColumnName` to column name in metadata, sent by server. The normalised literal, that is stored internally, should be the same at the end.